cmd/init.go

```diff
@@ -51,6 +51,8 @@ func init() {
 	initCmd.PersistentFlags().StringArrayVar(&etcdAdmConfig.EtcdDiskPriorities, "disk-priorities", constants.DefaultEtcdDiskPriorities, "Setting etcd disk priority")
 	initCmd.PersistentFlags().StringVar((*string)(&etcdAdmConfig.InitSystem), "init-system", string(apis.Systemd), "init system type (systemd or kubelet)")
 	initCmd.PersistentFlags().StringVar(&etcdAdmConfig.ImageRepository, "image-repository", constants.DefaultImageRepository, "image repository when using kubelet init system")
+	initCmd.PersistentFlags().StringVar(&etcdAdmConfig.DataDir, "data-dir", constants.DefaultDataDir, "etcd data directory")
```

Comment:
I like the way this PR has done it consistently with the other flags, but we should consider for a future PR a pattern I've seen elsewhere, where we call `etcdAdmConfig.InitDefaults()` and then use the value as the flag default

i.e. this would be 

```
initCmd.PersistentFlags().StringVar(&etcdAdmConfig.DataDir, "data-dir", etcdAdmConfig.DataDir, "etcd data directory")
```

It means we don't need to repeat the defaults (and can continue setting them in InitDefaults), but it does mean we can't have different defaults in init vs join, which may be while we did it this way here!